packages/ember-application/lib/system/application-instance.js

```diff
@@ -79,7 +87,9 @@ let ApplicationInstance = EmberObject.extend(RegistryProxy, ContainerProxy, {
 
     var application = get(this, 'application');
 
-    set(this, 'rootElement', get(application, 'rootElement'));
+    if (!isEnabled('ember-application-visit')) {
```

Comment:
They can't be created manually via public API, but they are used by Ember. Basically this is just trying to be carefully are preserve the /exact/ point where this was happening (I moved this exact call down into the boot method), I will be happy to unify them below if that's fine with you